File: lib/examples/website_linking_example.dart
import 'package:flutter/material.dart';
import '../widgets/website_link_button.dart';
import '../services/website_verification_service.dart';

class WebsiteLinkingExample extends StatefulWidget {
  @override
  _WebsiteLinkingExampleState createState() => _WebsiteLinkingExampleState();
}

class _WebsiteLinkingExampleState extends State<WebsiteLinkingExample> {
  final TextEditingController _domainController = TextEditingController();
  final int _currentUserId = 1; // Replace with actual user ID
  
  List<Map<String, dynamic>> _linkedWebsites = [];
  bool _isLoadingWebsites = false;

  @override
  void initState() {
    super.initState();
    _loadUserWebsites();
  }

  Future<void> _loadUserWebsites() async {
    setState(() => _isLoadingWebsites = true);
    
    try {
      final result = await WebsiteVerificationService.getUserWebsites(_currentUserId);
      
      if (result['success']) {
        setState(() {
          _linkedWebsites = List<Map<String, dynamic>>.from(result['websites']);
        });
      }
    } catch (e) {
      print('Error loading websites: $e');
    } finally {
      setState(() => _isLoadingWebsites = false);
    }
  }

  void _onLinkStatusChanged(bool isLinked, Map<String, dynamic>? websiteData) {
    if (isLinked) {
      _loadUserWebsites(); // Refresh the list
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Website Linking Demo'),
        backgroundColor: Colors.blue.shade600,
        foregroundColor: Colors.white,
      ),
      body: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            // Input section
            Card(
              child: Padding(
                padding: const EdgeInsets.all(16.0),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      'Link New Website',
                      style: TextStyle(
                        fontSize: 18,
                        fontWeight: FontWeight.bold,
                      ),
                    ),
                    const SizedBox(height: 16),
                    TextField(
                      controller: _domainController,
                      decoration: InputDecoration(
                        labelText: 'Domain',
                        hintText: 'example.com',
                        border: OutlineInputBorder(),
                        prefixIcon: Icon(Icons.language),
                      ),
                    ),
                    const SizedBox(height: 12),
                    WebsiteLinkButton(
                      domain: _domainController.text.trim(),
                      userId: _currentUserId,
                      onLinkStatusChanged: _onLinkStatusChanged,
                    ),
                  ],
                ),
              ),
            ),
            
            const SizedBox(height: 24),
            
            // Linked websites section
            Text(
              'Your Linked Websites',
              style: TextStyle(
                fontSize: 18,
                fontWeight: FontWeight.bold,
              ),
            ),
            const SizedBox(height: 12),
            
            if (_isLoadingWebsites)
              Center(child: CircularProgressIndicator())
            else if (_linkedWebsites.isEmpty)
              Container(
                padding: EdgeInsets.all(32),
                decoration: BoxDecoration(
                  color: Colors.grey.shade100,
                  borderRadius: BorderRadius.circular(8),
                ),
                child: Center(
                  child: Column(
                    children: [
                      Icon(
                        Icons.link_off,
                        size: 48,
                        color: Colors.grey.shade400,
                      ),
                      const SizedBox(height: 8),
                      Text(
                        'No websites linked yet',
                        style: TextStyle(
                          color: Colors.grey.shade600,
                          fontSize: 16,
                        ),
                      ),
                    ],
                  ),
                ),
              )
            else
              Expanded(
                child: ListView.builder(
                  itemCount: _linkedWebsites.length,
                  itemBuilder: (context, index) {
                    final website = _linkedWebsites[index];
                    return Card(
                      margin: const EdgeInsets.only(bottom: 8),
                      child: ListTile(
                        leading: CircleAvatar(
                          backgroundColor: Colors.blue.shade100,
                          child: Icon(
                            Icons.language,
                            color: Colors.blue.shade600,
                          ),
                        ),
                        title: Text(
                          website['website_name'] ?? 'Unknown Website',
                          style: TextStyle(fontWeight: FontWeight.bold),
                        ),
                        subtitle: Text(website['domain'] ?? ''),
                        trailing: Container(
                          padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
                          decoration: BoxDecoration(
                            color: Colors.green.shade100,
                            borderRadius: BorderRadius.circular(12),
                          ),
                          child: Text(
                            'Linked',
                            style: TextStyle(
                              color: Colors.green.shade700,
                              fontSize: 12,
                              fontWeight: FontWeight.bold,
                            ),
                          ),
                        ),
                        onTap: () {
                          // Navigate to website products or details
                          print('Tapped on website: ${website['website_id']}');
                        },
                      ),
                    );
                  },
                ),
              ),
          ],
        ),
      ),
    );
  }
}

File: lib/server/examples/WebsiteLinkingExample.js
import React, { useState, useEffect } from 'react';
import axios from 'axios';
import WebsiteLinkButton from '../components/WebsiteLinkButton';

const WebsiteLinkingExample = () => {
  const [domain, setDomain] = useState('');
  const [linkedWebsites, setLinkedWebsites] = useState([]);
  const [isLoadingWebsites, setIsLoadingWebsites] = useState(false);
  const currentUserId = 1; // Replace with actual user ID

  useEffect(() => {
    loadUserWebsites();
  }, []);

  const loadUserWebsites = async () => {
    setIsLoadingWebsites(true);
    
    try {
      const response = await axios.get(`/api/websites/user/${currentUserId}`);
      
      if (response.data.success) {
        setLinkedWebsites(response.data.data || []);
      }
    } catch (error) {
      console.error('Error loading websites:', error);
    } finally {
      setIsLoadingWebsites(false);
    }
  };

  const onLinkStatusChanged = (isLinked, websiteData) => {
    if (isLinked) {
      loadUserWebsites(); // Refresh the list
    }
  };

  return (
    <div className="min-h-screen bg-gray-50 py-8">
      <div className="max-w-4xl mx-auto px-4">
        <h1 className="text-3xl font-bold text-gray-900 mb-8">Website Linking Demo</h1>
        
        {/* Input Section */}
        <div className="bg-white rounded-lg shadow-md p-6 mb-8">
          <h2 className="text-xl font-bold text-gray-800 mb-6">Link New Website</h2>
          
          <div className="space-y-4">
            <div>
              <label className="block text-sm font-medium text-gray-700 mb-2">
                Domain
              </label>
              <div className="relative">
                <div className="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
                  <svg className="h-5 w-5 text-gray-400" fill="currentColor" viewBox="0 0 20 20">
                    <path fillRule="evenodd" d="M12.586 4.586a2 2 0 112.828 2.828l-3 3a2 2 0 01-2.828 0 1 1 0 00-1.414 1.414 4 4 0 005.656 0l3-3a4 4 0 00-5.656-5.656l-1.5 1.5a1 1 0 101.414 1.414l1.5-1.5zm-5 5a2 2 0 012.828 0 1 1 0 101.414-1.414 4 4 0 00-5.656 0l-3 3a4 4 0 105.656 5.656l1.5-1.5a1 1 0 10-1.414-1.414l-1.5 1.5a2 2 0 11-2.828-2.828l3-3z" clipRule="evenodd" />
                  </svg>
                </div>
                <input
                  type="text"
                  value={domain}
                  onChange={(e) => setDomain(e.target.value)}
                  placeholder="example.com"
                  className="pl-10 block w-full border border-gray-300 rounded-md py-2 px-3 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-blue-500"
                />
              </div>
            </div>
            
            <div className="pt-4">
              <WebsiteLinkButton
                domain={domain}
                userId={currentUserId}
                onLinkStatusChanged={onLinkStatusChanged}
              />
            </div>
          </div>
        </div>
        
        {/* Linked Websites Section */}
        <div className="bg-white rounded-lg shadow-md p-6">
          <h2 className="text-xl font-bold text-gray-800 mb-6">Your Linked Websites</h2>
          
          {isLoadingWebsites ? (
            <div className="flex justify-center py-8">
              <div className="animate-spin rounded-full h-8 w-8 border-b-2 border-blue-600"></div>
            </div>
          ) : linkedWebsites.length === 0 ? (
            <div className="text-center py-12 bg-gray-50 rounded-lg">
              <svg className="mx-auto h-12 w-12 text-gray-400 mb-4" fill="currentColor" viewBox="0 0 20 20">
                <path fillRule="evenodd" d="M12.586 4.586a2 2 0 112.828 2.828l-3 3a2 2 0 01-2.828 0 1 1 0 00-1.414 1.414 4 4 0 005.656 0l3-3a4 4 0 00-5.656-5.656l-1.5 1.5a1 1 0 101.414 1.414l1.5-1.5zm-5 5a2 2 0 012.828 0 1 1 0 101.414-1.414 4 4 0 00-5.656 0l-3 3a4 4 0 105.656 5.656l1.5-1.5a1 1 0 10-1.414-1.414l-1.5 1.5a2 2 0 11-2.828-2.828l3-3z" clipRule="evenodd" />
              </svg>
              <p className="text-gray-600 text-lg">No websites linked yet</p>
            </div>
          ) : (
            <div className="space-y-3">
              {linkedWebsites.map((website) => (
                <div key={website.website_id} className="border border-gray-200 rounded-lg p-4 hover:bg-gray-50 transition-colors">
                  <div className="flex items-center justify-between">
                    <div className="flex items-center space-x-4">
                      <div className="flex-shrink-0">
                        <div className="w-10 h-10 bg-blue-100 rounded-full flex items-center justify-center">
                          <svg className="w-6 h-6 text-blue-600" fill="currentColor" viewBox="0 0 20 20">
                            <path fillRule="evenodd" d="M12.586 4.586a2 2 0 112.828 2.828l-3 3a2 2 0 01-2.828 0 1 1 0 00-1.414 1.414 4 4 0 005.656 0l3-3a4 4 0 00-5.656-5.656l-1.5 1.5a1 1 0 101.414 1.414l1.5-1.5zm-5 5a2 2 0 012.828 0 1 1 0 101.414-1.414 4 4 0 00-5.656 0l-3 3a4 4 0 105.656 5.656l1.5-1.5a1 1 0 10-1.414-1.414l-1.5 1.5a2 2 0 11-2.828-2.828l3-3z" clipRule="evenodd" />
                          </svg>
                        </div>
                      </div>
                      <div>
                        <h3 className="text-lg font-semibold text-gray-900">
                          {website.website_name || 'Unknown Website'}
                        </h3>
                        <p className="text-gray-600">{website.domain}</p>
                      </div>
                    </div>
                    <div className="flex items-center space-x-2">
                      <span className="inline-flex items-center px-3 py-1 rounded-full text-xs font-medium bg-green-100 text-green-800">
                        Linked
                      </span>
                      <button
                        onClick={() => {
                          // Navigate to website products or details
                          console.log('Tapped on website:', website.website_id);
                        }}
                        className="text-blue-600 hover:text-blue-800 font-medium text-sm"
                      >
                        View Products →
                      </button>
                    </div>
                  </div>
                </div>
              ))}
            </div>
          )}
        </div>
      </div>
    </div>
  );
};

export default WebsiteLinkingExample;
